libmcount/mcount.c

```diff
@@ -479,11 +483,13 @@ void mcount_entry_filter_record(struct mcount_thread_data *mtdp,
 			}
 		}
 
+#ifdef HAVE_LIBPYTHON2
 		/* script hooking for function entry */
 		if (script_str) {
 			rstack->tid = gettid(mtdp);
 			script_uftrace_entry(rstack);
 		}
+#endif
```

Comment:
Hmm.. it'd be nice if compier could optimize out the whole if statement if possible.  If gcc defines the `HAVE_LIBPYTHON2` as value of 1, how about calling it like below?

```
if (HAVE_LIBPYTHON2 && script_str) {
    rstack->tid = gettid(mtdp);
    script_uftrace_entry(rstack);
}
```